src/lib/utils/series-utils.js

```diff
@@ -65,6 +65,42 @@ function collectSeriesTypesInfo(children) {
   return result;
 }
 
+export function getStackedData(children, attr) {
+  const childData = [];
+  const attr0 = `${attr}0`;
+  children.forEach((child, childIndex) => {
+    if (!child) {
+      childData.push(null);
+      return;
+    }
+
+    const {data} = child.props;
+    if (!attr || !data || !data.length) {
+      childData.push(data);
+      return;
+    }
+    childData.push(data.map((d, dIndex) => {
+      let prevValue = 0;
+      let prevIndex = childIndex - 1;
+      while (prevIndex >= 0) {
+        if (childData[prevIndex]) {
+          break;
+        }
+        prevIndex--;
+      }
```

Comment:
This seems quite confusing, you're looping through the children, then looping to find the previous child that is truthy?

This logic to find the `prevIndex` seems like it's something that could be outside of the `data.map` loop. You're trying to find the previous child's data that is truthy right?